python/across_api/burstcube/toorequest.py

```diff
@@ -0,0 +1,437 @@
+        if self.skycoord is not None or self.healpix_loc is not None:
+            fov = BurstCubeFOV(ephem=ephem, time=self.trigger_time)
+            infov = fov.probability_in_fov(
+                skycoord=self.skycoord,
+                error_radius=self.error_radius,
+                healpix_loc=self.healpix_loc,
+                healpix_scheme=self.healpix_scheme,
+            )
+
+            # Flag up if the required probability is not met
+            if infov < self.min_prob:
+                self.too_info += f"Probability inside FOV: {100*infov:.2f}%. Trigger was occulted at T0. "
+                self.reject_reason = TOOReason.earth_occult
+                return False
+            else:
+                self.too_info += f"Probability inside FOV: {100*infov:.2f}%. "
+
+        # Check if any part of the dump time is inside the SAA, warn if so
+        if True in burstcube_saa_constraint(time=ephem.timestamp, ephem=ephem):
+            self.too_info += "Dump time partially inside SAA."
+
+        #  Strip excess whitespace
+        self.too_info = self.too_info.strip()
+        return True
+
+    def post(self) -> bool:
+        """
+        Upload BurstCubeTOO to ACROSS API using POST
+
+        Returns
+        -------
+            Did this work? True | False
+        """
+        # Shouldn't run this unless a created_by is set
+        if self.username is None:
+            raise HTTPException(401, "Username not set.")
+
+        # Validate supplied BurstCubeTOO values against the Schema
+        if not self.validate_post():
+            return False
+
+        # Set the created_on and created_by fields
+        self.created_on = Time.now()
+        self.created_by = self.username
+
+        # Set the ID of this TOO
+        self.id = hashlib.md5(
+            f"{self.trigger_time}{self.exposure}{self.offset}".encode()
+        ).hexdigest()
+
+        # Check if this BurstCubeTOO exists. As the id is just a hash of the
+        # trigger_time, exposure and offset, then repeated requests for values
+        # that match this will be caught.
+        try:
+            response = self.table.get_item(Key={"id": self.id})
+            if response.get("Item"):
+                raise HTTPException(409, "BurstCubeTOO already exists.")
+        except botocore.exceptions.ClientError:
+            pass
+
+        # Check for various TOO constraints
+        if not self.check_constraints():
+            self.status = TOOStatus.rejected
+
+        # Write BurstCubeTOO to the database
+        json_too = json.loads(self.schema.model_dump_json(), parse_float=Decimal)
+        self.table.put_item(Item=json_too)
+
+        return True
+
+
+class BurstCubeTOORequests(ACROSSAPIBase):
+    """
+    Class to fetch multiple BurstCubeTOO requests, based on various filters.
+
+    Note that the filtering right now is based on DynamoDB scan, which is not
+    very efficient. This should be replaced with a query at some point.
```

Comment:
Indeed, that's both inefficient and expensive. Please make the time the sort key of the table and use the `<`, `<=`, `>`, `>=`, or `BETWEEN` operators in the [`KeyConditionExpression`](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html#DDB-Query-request-KeyConditionExpression).

Reply:
I implemented this using a GSI (see checking [7e476a7](https://github.com/nasa-gcn/gcn.nasa.gov/pull/1975/commits/7e476a750d1bb78f9273772c772b5068b4c7924a))

Reply:
Do you need the dummy partition key `gsipk`? Could you instead use the id as the partition key for the secondary index?

Reply:
AFAIK you have to give a partition key value for a query where you're filtering on the sort key. You can only do `between` on a sort key, so I have to define a GSI PK that is constant across the whole database. If there's another way to do this, let me know.